src/app/replayer/replayer.ml

```diff
@@ -340,30 +474,46 @@ let main ~input_file ~output_file ~archive_uri () =
           ~depth:constraint_constants.ledger_depth padded_accounts
       in
       let ledger = Lazy.force @@ Genesis_ledger.Packed.t packed_ledger in
-      let state_hash =
-        State_hash.to_yojson input.target_state_hash
-        |> unquoted_string_of_yojson
+      let epoch_ledgers_state_hash =
+        State_hash.to_string input.target_epoch_ledgers_state_hash
+      in
+      [%log info] "Retrieving fork block state_hash" ;
+      let%bind fork_state_hash =
+        state_hash_of_epoch_ledgers_state_hash ~logger pool
+          epoch_ledgers_state_hash
+      in
+      [%log info] "Loading epoch ledger data" ;
+      let%bind {staking_epoch_data_id; next_epoch_data_id} =
+        epoch_data_ids_of_state_hash ~logger pool fork_state_hash
+      in
+      let%bind { epoch_data_hash= staking_epoch_ledger_hash_str
+               ; epoch_data_seed= staking_seed_str } =
+        epoch_data_of_id ~logger pool staking_epoch_data_id
+      in
+      let%bind { epoch_data_hash= next_epoch_ledger_hash_str
+               ; epoch_data_seed= next_seed_str } =
+        epoch_data_of_id ~logger pool next_epoch_data_id
+      in
+      let staking_epoch_ledger_hash =
+        Frozen_ledger_hash.of_string staking_epoch_ledger_hash_str
+      in
+      let staking_seed = Epoch_seed.of_string staking_seed_str in
+      let next_epoch_ledger_hash =
+        Frozen_ledger_hash.of_string next_epoch_ledger_hash_str
       in
-      [%log info] "Loading block information" ;
+      let next_seed = Epoch_seed.of_string next_seed_str in
```

Comment:
`next_seed` should be from `next_epoch_data` of the fork state hash. Explained in this comment why that needs to be the case https://github.com/MinaProtocol/mina/pull/6620#issuecomment-723361981

Reply:
Oops, will fix.